src/test/munit/sftp-delete-directory-with-config-file-negative-test-case.xml

```diff
@@ -0,0 +1,57 @@
+<?xml version="1.0" encoding="UTF-8"?>
+
+<mule xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xmlns:munit="http://www.mulesoft.org/schema/mule/munit"
+      xmlns:munit-tools="http://www.mulesoft.org/schema/mule/munit-tools"
+      xmlns="http://www.mulesoft.org/schema/mule/core"
+      xmlns:sftp="http://www.mulesoft.org/schema/mule/sftp"
+      xmlns:java="http://www.mulesoft.org/schema/mule/java" xmlns:test="http://www.mulesoft.org/schema/mule/test"
+      xsi:schemaLocation="
+		http://www.mulesoft.org/schema/mule/core http://www.mulesoft.org/schema/mule/core/current/mule.xsd
+		http://www.mulesoft.org/schema/mule/munit http://www.mulesoft.org/schema/mule/munit/current/mule-munit.xsd
+		http://www.mulesoft.org/schema/mule/munit-tools  http://www.mulesoft.org/schema/mule/munit-tools/current/mule-munit-tools.xsd
+        http://www.mulesoft.org/schema/mule/sftp http://www.mulesoft.org/schema/mule/sftp/current/mule-sftp.xsd
+        http://www.mulesoft.org/schema/mule/java http://www.mulesoft.org/schema/mule/java/current/mule-java.xsd">
+
+    <munit:config name="sftp-delete-directory-with-config-file-test-case.xml">
+        <munit:parameterizations>
+            <munit:parameterization name="file-enabled-config">
+                <munit:parameters>
+                    <munit:parameter propertyName="config" value="file-enabled-config"/>
+                </munit:parameters>
+            </munit:parameterization>
+        </munit:parameterizations>
+    </munit:config>
+
+
+
+    <munit:before-suite name="before-sftp-delete-directory-test-case">
+        <java:invoke-static class="org.mule.extension.sftp.internal.lifecycle.SftpServerLifecycleManager"
+                            method="startSftpServer(String)">
+            <java:args>
+                #[{
+                arg0: ${sftp.server.port}
+                }]
+            </java:args>
+        </java:invoke-static>
+    </munit:before-suite>
+
+    <munit:after-suite name="after-sftp-delete-directory-test-case">
+        <java:invoke-static class="org.mule.extension.sftp.internal.lifecycle.SftpServerLifecycleManager"
+                            method="stopSftpServer()">
+            <java:args>
+                #[{}]
+            </java:args>
+        </java:invoke-static>
+    </munit:after-suite>
+
+
+    <munit:test name="sftp-delete-not-exist-directory" description="Test that tries to remove a non-existent directory, it expects an error SFTP:NEGOTIATION_FAILED"
+                expectedErrorType="SFTP:NEGOTIATION_FAILED">
```

Comment:
Why an operation trying to delete a non-existant directory would throw an error of NEGOTIATION_FAILED? That should be ILLEGAL_PATH or FILE_DOESNT_EXIST error.
I understood that NEGOTIATION_FAILED was a new kind of error for when the connection to the SFTP server was unsuccessful due to the cipher's mismatch.

Reply:
Refactored name and description.